R/xgboost-development.R

```diff
@@ -0,0 +1,237 @@
+#' Compare predictive models, created on your data
+#'
+#' @description This step allows you to create an XGBoost model, based on
+#' your data.
+#' @docType class
+#' @usage XGBoost Development(object, type, df, grainCol, predictedCol, 
+#' impute, debug)
+#' @import caret
+#' @import doParallel
+#' @import e1071
+#' @import grpreg
+#' @import pROC
+#' @importFrom R6 R6Class
+#' @import ranger
+#' @import ROCR
+#' @import RODBC
+#' @param object of SuperviseModelParameters class for $new() constructor
+#' @param type The type of model (either 'regression' or 'classification')
+#' @param df Dataframe whose columns are used for calc.
+#' @param grainCol Optional. The dataframe's column that has IDs pertaining to 
+#' the grain. No ID columns are truly needed for this step.
+#' @param predictedCol Column that you want to predict. If you're doing
+#' classification then this should be Y/N.
+#' @param impute Set all-column imputation to F or T.
+#' This uses mean replacement for numeric columns
+#' and most frequent for factorized columns.
+#' F leads to removal of rows containing NULLs.
+#' @param debug Provides the user extended output to the console, in order
+#' to monitor the calculations throughout. Use T or F.
+#' @references \url{http://hctools.org/}
+#' @seealso \code{\link{LassoDevelopment}}
+#' @seealso \code{\link{LinearMixedModelDevelopment}}
+#' @seealso \code{\link{healthcareai}}
+#' @examples
+#'
+#' #### Example using csv dataset ####
+#' This dataset comes from \url{http://archive.ics.uci.edu/ml/datasets/dermatology}
+#' ptm <- proc.time()
+#' library(healthcareai)
+#' 
+#' # 1. Load data. Categorical columns should be characters.
+#' csvfile <- system.file("extdata", 
+#'                       "dermatology_multiclass_data.csv", 
+#'                       package = "healthcareai")
+#' 
+#'  # Replace csvfile with 'path/file'
+#' df <- read.csv(file = csvfile, 
+#'               header = TRUE, 
+#'               stringsAsFactors = FALSE,
+#'               na.strings = c("NULL", "NA", "", "?"))
+#' 
+#' str(df)
```

Comment:
Maybe add comment above the str command along the lines of "Check type of columns"